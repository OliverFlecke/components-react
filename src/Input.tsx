import React, { forwardRef } from 'react';
import useId from './utils/useId';

export interface InputProps {
	label?: string;
	errorMessage?: string;
}

const Input = forwardRef<
	HTMLInputElement,
	InputProps & React.InputHTMLAttributes<HTMLInputElement>
>((props, ref) => {
	const { label, errorMessage, ...inputProps } = props;
	const id = useId('input');

	return (
		<div className="space-y-2">
			<label
				htmlFor={id}
				className="block text-gray-700 dark:text-gray-300 font-medium text-sm"
			>
				{label}
			</label>
			<input
				ref={ref}
				id={id}
				type="text"
				{...inputProps}
				className={`py-2 px-4 rounded-md shadow focus:outline-none focus:ring focus:border-indigo-400 bg-white dark:bg-gray-900 dark:text-gray-100 ${props.className}`}
			/>
			{errorMessage !== undefined && (
				<div className="text-red-700 dark:text-red-600 text-sm">
					{errorMessage}
				</div>
			)}
		</div>
	);
});

export default Input;
