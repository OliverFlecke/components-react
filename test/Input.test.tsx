import React from 'react';
import renderer from 'react-test-renderer';
import Input from '../src/Input';

describe('Input', () => {
	test('render snapshot', () => {
		const tree = renderer.create(<Input label="Email" />);
		expect(tree.toJSON()).toMatchSnapshot();
	});

	test('render snapshot with error message', () => {
		const tree = renderer.create(
			<Input label="Email" errorMessage="Invalid email address" />
		);
		expect(tree.toJSON()).toMatchSnapshot();
	});
});
