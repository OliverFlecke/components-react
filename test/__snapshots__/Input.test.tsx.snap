// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input render snapshot 1`] = `
<div
  className="space-y-2"
>
  <label
    className="space-y-2"
  >
    <span
      className="input-label"
    >
      Email
    </span>
    <input
      className="py-2 px-4 rounded-md shadow focus:outline-none focus:ring focus:border-indigo-400 bg-white dark:bg-gray-900 dark:text-gray-100"
      type="text"
    />
  </label>
</div>
`;

exports[`Input render snapshot with error message 1`] = `
<div
  className="space-y-2"
>
  <label
    className="space-y-2"
  >
    <span
      className="input-label"
    >
      Email
    </span>
    <input
      className="py-2 px-4 rounded-md shadow focus:outline-none focus:ring focus:border-indigo-400 bg-white dark:bg-gray-900 dark:text-gray-100"
      type="text"
    />
  </label>
  <div
    className="text-red-700 dark:text-red-600 text-sm"
  >
    Invalid email address
  </div>
</div>
`;
